mmv1/products/iamworkforcepool/WorkforcePoolProvider.yaml

```diff
@@ -252,6 +256,30 @@ properties:
           The client ID. Must match the audience claim of the JWT issued by the
           identity provider.
         required: true
+      - !ruby/object:Api::Type::NestedObject
+        name: 'clientSecret'
+        description: |
+          The optional client secret. Required to enable Authorization Code flow for web sign-in.
+        properties:
+          - !ruby/object:Api::Type::NestedObject
+            name: 'value'
+            exactly_one_of:
+              - oidc.0.client_secret.0.value
+            description: |
+              The value of the client secret.
+            properties:
+              - !ruby/object:Api::Type::String
+                name: 'plainText'
+                description: |
+                  The plain text of the client secret value.
+                sensitive: true
+                required: true
+                custom_flatten: templates/terraform/custom_flatten/iam_workforce_pool_provider_oidc_client_secret_value.go.erb
```

Comment:
Can we use ignore_read here instead of a custom flattener?

Reply:
I tried that and unfortunately it does not work. I think ignore_read does not work with nested fields: https://github.com/GoogleCloudPlatform/magic-modules/pull/7143#discussion_r1080660150.